contrib/rust/src/python/pants/contrib/rust/tasks/cargo_workspace.py

```diff
@@ -0,0 +1,144 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from future.utils import text_type
+from pants.base.build_environment import get_buildroot
+from pants.engine.fs import PathGlobs, PathGlobsAndRoot
+from pants.source.wrapped_globs import EagerFilesetWithSpec, RGlobs
+
+from pants.contrib.rust.targets.synthetic.cargo_project_binary import CargoProjectBinary
+from pants.contrib.rust.targets.synthetic.cargo_project_library import CargoProjectLibrary
+from pants.contrib.rust.targets.synthetic.cargo_project_test import CargoProjectTest
+from pants.contrib.rust.targets.synthetic.cargo_synthetic_binary import CargoSyntheticBinary
+from pants.contrib.rust.targets.synthetic.cargo_synthetic_custom_build import \
+  CargoSyntheticCustomBuild
+from pants.contrib.rust.targets.synthetic.cargo_synthetic_library import CargoSyntheticLibrary
+from pants.contrib.rust.tasks.cargo_task import CargoTask
+
+
+class Workspace(CargoTask):
```

Comment:
If this class is intended to be abstract, it would be good to have it subclass `AbstractClass` as well:
```
from pants.util.meta import AbstractClass
...
class NativeCompile(NativeTask, AbstractClass):
```